velox/functions/prestosql/InPredicate.cpp

```diff
@@ -335,47 +445,20 @@ class InPredicate : public exec::VectorFunction {
   }
 
   static std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
-    // boolean|tinyint|smallint|integer|bigint|varchar... -> boolean
-    std::vector<std::shared_ptr<exec::FunctionSignature>> signatures;
-    for (auto& type :
-         {"boolean",
-          "tinyint",
-          "smallint",
-          "integer",
-          "bigint",
-          "varchar",
-          "varbinary",
-          "double",
-          "real",
-          "date"}) {
-      signatures.emplace_back(exec::FunctionSignatureBuilder()
-                                  .returnType("boolean")
-                                  .argumentType(type)
-                                  .argumentType(fmt::format("array({})", type))
-                                  .build());
-      signatures.emplace_back(exec::FunctionSignatureBuilder()
-                                  .returnType("boolean")
-                                  .argumentType(type)
-                                  .argumentType("array(unknown)")
-                                  .build());
-    }
-    signatures.emplace_back(
+    return {
         exec::FunctionSignatureBuilder()
+            .typeVariable("T")
             .returnType("boolean")
-            .integerVariable("precision")
-            .integerVariable("scale")
-            .argumentType("DECIMAL(precision, scale)")
-            .argumentType("array(DECIMAL(precision, scale))")
-            .build());
-    signatures.emplace_back(exec::FunctionSignatureBuilder()
-                                .returnType("boolean")
-                                .integerVariable("precision")
-                                .integerVariable("scale")
-                                .argumentType("DECIMAL(precision, scale)")
-                                .argumentType("array(unknown)")
-                                .build());
-
-    return signatures;
+            .argumentType("T")
+            .constantArgumentType("array(T)")
+            .build(),
+        exec::FunctionSignatureBuilder()
+            .typeVariable("T")
+            .returnType("boolean")
+            .argumentType("T")
+            .constantArgumentType("array(unknown)")
```

Comment:
this is to cover the "IN (NULL) " case, right?

Reply:
That's right.